tests/extensions/provisioning/defaultK8sVersion.go

```diff
@@ -0,0 +1,66 @@
+package provisioning
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/clusters/kubernetesversions"
+	"github.com/rancher/tfp-automation/config"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	RKE1    = "rke1"
+	RKE2    = "rke2"
+	K3S     = "k3s"
+	RANCHER = "-rancher"
+)
+
+// DefaultK8sVersion is a function that will set the default Kubernetes version if the user has not specified one.
+func DefaultK8sVersion(t *testing.T, client *rancher.Client, clusterConfig *config.TerratestConfig, terraformConfig *config.TerraformConfig) {
+	if clusterConfig.KubernetesVersion == "" {
+		if strings.Contains(terraformConfig.Module, RKE1) {
+			defaultVersion, err := kubernetesversions.Default(client, clusters.RKE1ClusterType.String(), nil)
+			require.NoError(t, err)
+
+			clusterConfig.KubernetesVersion = defaultVersion[0]
+		} else if strings.Contains(terraformConfig.Module, RKE2) {
+			defaultVersion, err := kubernetesversions.Default(client, clusters.RKE2ClusterType.String(), nil)
+			require.NoError(t, err)
+
+			clusterConfig.KubernetesVersion = defaultVersion[0]
+		} else if strings.Contains(terraformConfig.Module, K3S) {
+			defaultVersion, err := kubernetesversions.Default(client, clusters.K3SClusterType.String(), nil)
+			require.NoError(t, err)
+
+			clusterConfig.KubernetesVersion = defaultVersion[0]
+		}
+	}
+}
+
+// DefaultUpgradedK8sVersion is a function that will set the default Kubernetes upgrade version
+// if the user has not specified one.
+func DefaultUpgradedK8sVersion(t *testing.T, client *rancher.Client, clusterConfig *config.TerratestConfig, terraformConfig *config.TerraformConfig) {
```

Comment:
does this work since its just a pointer to the object `clusterConfig`? it feels a little weird having a nil return update something that the user is providing. Maybe we can update the name to something like `SetDefaultUpgradedK8SVersion` or something so that we know it will actually change something?

Reply:
Terraform only needs to acknowledge an increase in the K8s version to update the respective resource block. I'm not strongly opposed to that, but I generally am not a fan of having method names as getters and setters when only one is involved.